frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,323 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import {CatalogTileView} from 'patternfly-react-extensions/dist/esm/components/CatalogTileView';
+import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
+import {EmptyState} from 'patternfly-react/dist/esm/components/EmptyState';
+import FormControl from 'patternfly-react/dist/esm/components/Form/FormControl';
+
+import {normalizeIconClass} from '../catalog-item-icon';
+
+const getFilters = (items, filterCategories=[]) => {
+  const filters = {
+    byName: {
+      value: '',
+      active: 'false',
+    },
+  };
+
+  _.each(items, item => {
+    if (filterCategories.length > 0) {
+      // Only consider filters in filterCategories
+      _.each(filterCategories, key => {
+        const tileKey = `tile${_.capitalize(key)}`;
+        const filterKey = `by${tileKey}`;
+        const tileValue = item[tileKey];
+        if (tileValue && (!filters[filterKey] || !filters[filterKey][tileValue]) ) {
+          if (!filters[filterKey]) {
+            filters[filterKey] = {};
+          }
+          filters[filterKey][tileValue] = {
+            label: tileValue,
+            value: tileValue,
+            active: false,
+          };
+        }
+      });
+    } else {
+      // Find filters from item objects
+      _.forOwn(item, (value, key) => {
+        if (key === 'obj' || key.includes('IconClass') || key.includes('ImgUrl')) {
+          return;
+        }
+        const filterKey = `by${key}`;
+        if (value && (!filters[filterKey] || !filters[filterKey][value]) ) {
+          if (!filters[filterKey]) {
+            filters[filterKey] = {};
+          }
+          filters[filterKey][value] = {
+            label: value,
+            value: value,
+            active: false,
+          };
+        }
+      });
+    }
+  });
+
+  return filters;
+};
+
+const filterCategories = [
+  'provider',
+];
+
+export class MarketplaceTileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const {items} = this.props;
+
+    const filters = getFilters(items, filterCategories);
+
+    const filterCounts = this.getFilterCounts(filters);
+
+    this.state = {
+      filteredItems: items,
+      filters,
+      filterCounts,
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { filters } = this.state;
+    const { items } = this.props;
+
+    // Update filters to match new items
+    if (items !== prevProps.items) {
+      const newFilters = getFilters(items, filterCategories);
+      this.setState({filters: newFilters});
+    }
+
+    // Update tile display and filter counts according to active filters
+    if (filters !== prevState.filters) {
+      this.setState({...this.getFilterItemsAndCounts(filters)});
+    }
```

Comment:
Should only call `setState` once. Would prefer to see the use of `getDerivedStateFromProps` to avoid multiple renders.

Reply:
Will do. I was following the implementation for Catalog's FilterSidePanel, but I'll use `getDerivedStateFromProps`